### ISSUE TITLE
process 1329 stopped error

### ISSUE THREAD

**Issue by fcoremedi** [aesophor/havoc.cc#2]
Process 1329 stopped
* thread #1, name = 'MainThrd', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
solution? not working in Mac Studio M1 Max

**Comment by aesophor** [comment]
MacBook Pro M1 Pro

That `SIGSTOP` message means the CSGO process has been temporarily stopped by `lldb` (LLVM debugger) for [process injection](https://www.picussecurity.com/resource/t1055-process-injection-of-the-mitre-attck-framework#:~:text=Process%20injection%20is%20a%20technique,network%20resources%2C%20and%20elevated%20privileges.). The `./inject.sh` uses `dlopen()` to load the cheat (as a shared library) into the address space of the CSGO process, hijack some functions in CSGO, and finally let CSGO resume.

```
~/Code/Personal/havoc.cc (master) » ./inject.sh                                                                                                                                         aesophor@Ludibrium
(lldb) attach --name csgo_osx64
Process 36482 stopped
* thread #1, name = 'MainThrd', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    frame #0: 0x00007ffa1f5758d6 AGXMetalG13X`___lldb_unnamed_symbol4108 + 2326
AGXMetalG13X`___lldb_unnamed_symbol4108:
->  0x7ffa1f5758d6 <+2326>: mov    dword ptr [rax + 4*rdx], esi
    0x7ffa1f5758d9 <+2329>: sub    rbx, r8
    0x7ffa1f5758dc <+2332>: and    rbx, r8
    0x7ffa1f5758df <+2335>: add    rdi, 0x8
Target 0: (csgo_osx64) stopped.
Executable module set to "/Users/aesophor/Library/Application Support/Steam/steamapps/common/Counter-Strike Global Offensive/csgo_osx64".
Architecture set to: x86_64-apple-macosx-.
(lldb) expr void* $library = (void*)dlopen("/Users/aesophor/Code/Personal/havoc.cc/build/libHavoc.dylib", 10)
(lldb) detach
Process 36482 detached
(lldb) quit
```

After injecting the cheat, go back to CSGO again and you should see this floating GUI menu.

<img width="1512" alt="Screenshot 2023-01-24 at 8 25 48 AM" src="https://user-images.githubusercontent.com/32998585/214184138-3430639c-2cce-42f7-ae2f-1d12e7c6c5b5.png">

**Comment by fcoremedi** [comment]
![Captura de pantalla 2023-01-24 a la(s) 01 17 09](https://user-images.githubusercontent.com/117494050/214211225-b91d0183-654d-4c85-9eb5-cb5417955260.png)

I do this:
- Launch CSGO
- Minimize the csgo
- Open terminal
-  Add file inject.sh to the terminal
- Press enter button
- Password User
- I inject the file and I see this terminal image))
- then back to my csgo and there's no floating GUI Menu.

what am i doing wrong?

**Comment by aesophor** [comment]
It's path problem

```
(void*)dlopen("/Users/fcoremedi/build/libHavoc.dylib", 10)
```

I think it should be
```
(void*)dlopen("/Users/fcoremedi/Downloads/havoc/build/libHavoc.dylib", 10)
```

If you type the following commands one by one, then the above problem should go away.
```
cd
git clone git@github.com:aesophor/havoc.cc.git
cd havoc.cc
mkdir -p build
cd build
cmake ..
make
cd ..
# Launch csgo from steam, and then come back to your shell
./inject.sh
```

**Comment by fcoremedi** [comment]
yeah, im trying to execute again this commands in terminal but I have this problem:

fatal: destination path 'havoc.cc' already exists and is not an empty directory.
fcoremedi@Mac-Studio-de-Francisco ~ % 

bro if you can give me your discord, gmail or something to help me it would be great :(
I'm new in this script world... hope you understand...
please!
this is my discord fcoremedi#4121

**Comment by fcoremedi** [comment]
Cloning into 'havoc.cc'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

**Comment by aesophor** [comment]
```
# Remove that directory so that you can clone it again
rm -rf havoc.cc
# Clone it without ssh so that no more permission denied (publickey)
git clone https://github.com/aesophor/havoc.cc
```

**Comment by fcoremedi** [comment]
okay bro now i have only a problem with this 
fcoremedi@Mac-Studio-de-Francisco build % cmake ..
zsh: command not found: cmake

**Comment by aesophor** [comment]
brew install cmake

**Comment by aesophor** [comment]
If it says command not found: brew

then install brew with the following command. [ref](https://brew.sh/)
```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

**Comment by fcoremedi** [comment]
![Captura de pantalla 2023-01-24 a la(s) 02 43 08](https://user-images.githubusercontent.com/117494050/214220482-614fecfa-9c64-40a1-b73a-ed3328c66512.png)

now im here XD

**Comment by aesophor** [comment]
Please follow the commands I gave you, carefully. You missed a `cd` command.

```
cd
git clone git@github.com:aesophor/havoc.cc.git
cd havoc.cc <--- you missed this one
mkdir -p build
cd build
cmake ..
make
cd ..
# Launch csgo from steam, and then come back to your shell
./inject.sh
```

**Comment by fcoremedi** [comment]
![Captura de pantalla 2023-01-24 a la(s) 03 23 47](https://user-images.githubusercontent.com/117494050/214225420-282dee84-eaab-448d-a072-530f76a8451b.png)

**Comment by aesophor** [comment]
I think this is the final part.

1. Since we are compiling the cheat as a x86_64 shared library and then run it with rosetta2 on arm64, we'll need to install `sdl2` library with the x86_64 homebrew instead of the arm64 homebrew. Follow this article: https://gist.github.com/progrium/b286cd8c82ce0825b2eb3b0b3a0720a0
2. After step 1, you'll get two versions of homebrew sitting simultaneously on your mac:
   - x86_64 homebrew
   - arm64 homebrew
3. Run
    - `arch -x86_64 /usr/local/bin/brew sdl2`
4. Continue from the `cmake ..` command (where you left off)
